app/javascript/mastodon/features/follow_requests/components/account_authorize.js

```diff
@@ -23,10 +24,43 @@ export default class AccountAuthorize extends ImmutablePureComponent {
     intl: PropTypes.object.isRequired,
   };
 
+  state = {
+    requestState: 'requested',
+  };
+
+  onClickAuthorize = () => {
+    this.props.onAuthorize();
+    this.setState({
+      requestState: 'authorized',
+    });
+  }
+
+  onClickReject = () => {
+    this.props.onReject();
+    this.setState({
+      requestState: 'rejected',
+    });
+  }
+
   render () {
-    const { intl, account, onAuthorize, onReject } = this.props;
+    const { intl, account } = this.props;
+    const { requestState } = this.state;
     const content = { __html: account.get('note_emojified') };
 
+    const accountPanel = () => {
+      const disabled = requestState !== 'requested';
+      return (
+        <div className='account--panel'>
+          <IconButton className={`account--panel__button ${requestState === 'authorized' ? requestState : ''}`} title={intl.formatMessage(messages.authorize)} icon='check' onClick={this.onClickAuthorize} disabled={disabled} />
+          <IconButton className={`account--panel__button ${requestState === 'rejected' ? requestState : ''}`} title={intl.formatMessage(messages.reject)} icon='times' onClick={this.onClickReject} disabled={disabled} />
+        </div>
+      );
+    };
+
+    if (!account) {
```

Comment:
`account` is marked required but it seems for a case the property is missing. Which is correct?

Reply:
Oh, it's my carelessness. This part is redundant.

Reply:
By the way, Account has the same code block for the case you pointed. I've not modified this, but is it better to remove...? 
https://github.com/tootsuite/mastodon/blob/3987bd18a4bfaee49ea9a3439fc51610aeca6c2c/app/javascript/mastodon/components/account.js#L58-L60

Reply:
You are right. But it is out of scope of this pull request. It should be handled in a different pull request.